biogeochem/EDPatchDynamicsMod.F90

```diff
@@ -2109,9 +2125,27 @@ subroutine create_patch(currentSite, new_patch, age, areap, label,nocomp_pft)
     call new_patch%tveg24%InitRMean(fixed_24hr,init_value=temp_init_veg,init_offset=real(hlm_current_tod,r8) )
     allocate(new_patch%tveg_lpa)
     call new_patch%tveg_lpa%InitRmean(ema_lpa,init_value=temp_init_veg)
+    allocate(new_patch%seedling_layer_par24)
+    call new_patch%seedling_layer_par24%InitRMean(fixed_24hr,init_value=init_seedling_par, init_offset=real(hlm_current_tod,r8))
+    allocate(new_patch%sdlng_mort_par)
+    call new_patch%sdlng_mort_par%InitRMean(ema_sdlng_mort_par,init_value=temp_init_veg)
+    allocate(new_patch%sdlng2sap_par)
+    call new_patch%sdlng2sap_par%InitRMean(ema_sdlng2sap_par,init_value=init_seedling_par)
+    
+    allocate(new_patch%sdlng_mdd(maxpft))
+    allocate(new_patch%sdlng_emerg_smp(maxpft))
+
+    do pft = 1,maxpft
+      allocate(new_patch%sdlng_mdd(pft)%p)
+      call new_patch%sdlng_mdd(pft)%p%InitRMean(ema_sdlng_mdd, init_value=0.0_r8)
+      allocate(new_patch%sdlng_emerg_smp(pft)%p)
+      call new_patch%sdlng_emerg_smp(pft)%p%InitRMean(ema_sdlng_emerg_h2o,init_value=init_seedling_smp)
```

Comment:
We used a similar structure to accommodate classes and class bound procedures with the fates-hydro water transfer functions.  I'm wondering though if this is just a lot of code to do something simple.  The exponential moving average is a pretty simple to apply mathematically, and we don't always have to use the running mean toolset to do it. That toolset is most useful for things like 24 hour window averages anyway.

Reply:
I used this code infrastructure partially because I needed a way to ensure that the moving averages would persist across timesteps, patch fission and fusion. Although the math of the ema is simple, these other aspects seem more complicated to implement and the existing infrastructure gave me something to follow. But let me know if you have a suggestion on a more elegant way to do this and I could give it a try!